### DISCUSSION TITLE
[Idea] save cpu cycles by running ci tests with --stop-on-failure

### DISCUSSION THREAD

**Discussion by tswfi** [PrestaShop/PrestaShop#37634] [Ideas]
### The problem

Integration tests run all the time for each pull request and for 5 (at the moment) different php versions. Running the tests takes quite a long time.

### The solution

Stopping the test on the first failure could save considerable amount of cpu cycles.

https://github.com/PrestaShop/PrestaShop/compare/develop...tswfi:PrestaShop:save_cpu_cycles_if_test_fails_stop_the_run?expand=1

Maybe even do it as a default by adding `stopOnFailure="true"` to the phpunit.xml files..

### Alternatives

more documentation for one time contributors to easily run the tests (and preferably faster than now)

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by tswfi** [comment]
phpunit documentation says that stop-on-error might also be needed but I have not tested those.. maybe someone who is well versed in phpunit can chime in on what would be the best way to do this (https://docs.phpunit.de/en/11.0/textui.html#automatically-stop-when)

**Comment by Hlavtox** [comment]
Downside of this approach is that you have to rerun the test each time after fixing one failure, instead of just fixing everything at once, possibly.

**Reply by tswfi** [reply]
True, but this with added documentation and an easy way to run the integration tests locally might still make the process cleaner and faster while saving time from the actions workflow.
